plugins/inventory/foreman.py

```diff
@@ -238,10 +296,203 @@ def _get_hostvars(self, host, vars_prefix='', omitted_vars=()):
                 hostvars[vars_prefix + k] = v
         return hostvars
 
+    def _fetch_params(self):
+        options = ("no", "yes")
+        params = dict()
+        report_options = {'want_location': True, 'want_organization': True, 'want_ipv4': True, 'want_ipv6': True,
+                          'want_host_group': True, 'want_hostcollections': False, 'want_subnet': True, 'want_subnet_v6': True, 'want_smart_proxies': True,
+                          'want_content_facet_attributes': False, 'want_host_params': True}
+        report_options.update(self.get_option('report') or {})
+
+        self.want_location = report_options['want_location']
+        self.want_organization = report_options['want_organization']
+        self.want_IPv4 = report_options['want_ipv4']
+        self.want_IPv6 = report_options['want_ipv6']
+        self.want_host_group = report_options['want_host_group']
+        self.want_hostcollections = report_options['want_hostcollections']
+        self.want_subnet = report_options['want_subnet']
+        self.want_subnet_v6 = report_options['want_subnet_v6']
+        self.want_smart_proxies = report_options['want_smart_proxies']
+        self.want_content_facet_attributes = report_options['want_content_facet_attributes']
+        self.want_host_params = report_options['want_host_params']
```

Comment:
Interestingly, I had to explicitly set that to `True` as otherwise it didn't fetch the params for me

Reply:
I ran it and I see host params..
```
"virt-who-fb1b6f1e.c091.42f4.bd69.fc1bcdcd0dae-1": {
                "foreman_content_attributes": null, 
                "foreman_host_group": null, 
                "foreman_host_parameters": {
                    "remote_execution_connect_by_ip": false, 
                    "remote_execution_effective_user_method": "sudo", 
                    "remote_execution_ssh_user": "root"
                }, 
                "foreman_id": 11, 
                "foreman_organization": "Default Organization", 
                "foreman_smart_proxies": [], 
                "foreman_subnet": null, 
                "foreman_subnet6": null

```

Reply:
Hmm. Need to look further. I was missing the domain name.

Reply:
Right, so when I don't set it, I get:

```json
            "testhost1.example.com": {
                "foreman_host_group": "group_a",
                "foreman_host_parameters": {
                    "remote_execution_connect_by_ip": false,
                    "remote_execution_effective_user_method": "sudo",
                    "remote_execution_ssh_user": "root",
                    "testparam1": "testvalue1",
                    "testparam2": "testvalue2"
                },
                "foreman_id": 1,
                "foreman_ipv4": null,
                "foreman_ipv6": null,
                "foreman_location": "Default Location",
                "foreman_organization": "Default Organization",
                "foreman_smart_proxies": [],
                "foreman_subnet": null,
                "foreman_subnet6": null
            },
```

and when I set it, I get:

```json
            "testhost1.example.com": {
                "foreman_architecture_id": null,
                "foreman_architecture_name": null,
                "foreman_build": false,
                "foreman_capabilities": [
                    "build"
                ],
                "foreman_certname": "testhost1.example.com",
                "foreman_comment": null,
                "foreman_compute_profile_id": null,
                "foreman_compute_profile_name": null,
                "foreman_compute_resource_id": null,
                "foreman_compute_resource_name": null,
                "foreman_created_at": "2020-11-10 10:01:13 UTC",
                "foreman_disk": null,
                "foreman_domain_id": 1,
                "foreman_domain_name": "example.com",
                "foreman_enabled": true,
                "foreman_environment_id": 1,
                "foreman_environment_name": "production",
                "foreman_global_status": 0,
                "foreman_global_status_label": "OK",
                "foreman_hostgroup_id": 1,
                "foreman_id": 1,
                "foreman_image_file": "",
                "foreman_image_id": null,
                "foreman_image_name": null,
                "foreman_installed_at": null,
                "foreman_ip": null,
                "foreman_ip6": null,
                "foreman_last_compile": null,
                "foreman_last_report": null,
                "foreman_location_id": 2,
                "foreman_location_name": "Default Location",
                "foreman_mac": null,
                "foreman_managed": false,
                "foreman_medium_id": null,
                "foreman_medium_name": null,
                "foreman_model_id": null,
                "foreman_model_name": null,
                "foreman_operatingsystem_id": 1,
                "foreman_operatingsystem_name": "CentOS 7.7",
                "foreman_organization_id": 1,
                "foreman_organization_name": "Default Organization",
                "foreman_owner_id": 4,
                "foreman_owner_name": "Admin User",
                "foreman_owner_type": "User",
                "foreman_provision_method": "build",
                "foreman_ptable_id": null,
                "foreman_ptable_name": null,
                "foreman_puppet_ca_proxy": null,
                "foreman_puppet_ca_proxy_id": null,
                "foreman_puppet_ca_proxy_name": null,
                "foreman_puppet_proxy": null,
                "foreman_puppet_proxy_id": null,
                "foreman_puppet_proxy_name": null,
                "foreman_puppet_status": 0,
                "foreman_pxe_loader": null,
                "foreman_realm_id": null,
                "foreman_realm_name": null,
                "foreman_sp_ip": null,
                "foreman_sp_mac": null,
                "foreman_sp_name": null,
                "foreman_sp_subnet_id": null,
                "foreman_subnet6_id": null,
                "foreman_subnet6_name": null,
                "foreman_subnet_id": null,
                "foreman_subnet_name": null,
                "foreman_updated_at": "2020-11-10 10:01:15 UTC",
                "foreman_uptime_seconds": null,
                "foreman_use_image": null,
                "foreman_uuid": null,
                "testparam1": "testvalue1",
                "testparam2": "testvalue2"
            },
```

Reply:
Aha. The one with actually failed back to non-reports.